modules/collection/programs/kitty.nix

```diff
@@ -0,0 +1,113 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}: let
+  inherit (lib.generators) mkKeyValueDefault;
+  inherit (lib.lists) hasPrefix;
+  inherit (lib.modules) mkIf;
+  inherit (lib.options) mkEnableOption mkOption mkPackageOption;
+  inherit (lib.strings) typeOf;
+  inherit (lib.types) oneOf nonEmptyStr submodule path;
+
+  keyValueSettings = {
+    listsAsDuplicateKeys = true;
+    mkKeyValue = mkKeyValueDefault {} " ";
+  };
+
+  keyValue = pkgs.formats.keyValue keyValueSettings;
+
+  themeType = oneOf [
+    path
+    nonEmptyStr
+  ];
+  lightDarkType = submodule {
+    options = {
+      light = mkOption {type = themeType;};
+      dark = mkOption {type = themeType;};
+    };
+  };
+  kittyThemeType = oneOf [
+    themeType
+    lightDarkType
+  ];
+
+  mkTheme = theme:
+    if ((typeOf theme) == "string") && !(hasPrefix "/" theme)
+    then "${pkgs.kitty-themes}/share/kitty-themes/themes/${theme}.conf"
+    else theme;
+
+  cfg = config.rum.programs.kitty;
+in {
+  options.rum.programs.kitty = {
+    enable = mkEnableOption "kitty";
+
+    package = mkPackageOption pkgs "kitty" {};
+
+    settings = mkOption {
+      type = keyValue.type;
+      default = {};
+      example = {
+        font_size = 10;
+        text_composition_strategy = "2.5 0";
+        cursor_shape = "beam";
+        map = [
+          "kitty_mod+l next_layout"
+          "ctrl+alt+t goto_layout tall"
+        ];
+      };
+      description = ''
+        The configuration converted and written to {file}`$HOME/.config/kitty/kitty.conf`.
+        Please reference https://sw.kovidgoyal.net/kitty/conf/ for config options.
+      '';
+    };
+
+    theme = mkOption {
+      type = kittyThemeType;
+      default = "";
+      example = {
+        light = "Modus_Vivendi";
+        dark = "Modus_Operandi";
+      };
+      description = ''
+        The theme(s) for kitty to use.
+        Previews can be found at https://github.com/dexpota/kitty-themes?tab=readme-ov-file#previews.
+
+        They can be either a name, if the theme is in `pkgs.kitty-themes`, or a path,
```

Comment:
I'm not a fan of this, but also based on the fact that `kitty-themes` is essentially just a big package of all packaged themes, this makes sense. I'm just unsure whether or not we should allow for such an abstraction? What about doing `${pkgs.kitty-themes}/share/kitty-themes/themes/Modus_Operandi.conf`? It is convenient, but I'm not sure if it's necessary. 

This is not blocking, I would just like your thoughts on it, completely up to you to leave it.